src/libraries/System.Private.CoreLib/src/System/Buffers/TlsOverPerCoreLockedStacksArrayPool.cs

```diff
@@ -292,58 +277,28 @@ public bool Trim()
         /// </remarks>
         private static bool Gen2GcCallbackFunc(object target)
         {
-            return ((TlsOverPerCoreLockedStacksArrayPool<T>)(target)).Trim();
-        }
-
-        private enum MemoryPressure
-        {
-            Low,
-            Medium,
-            High
-        }
-
-        private static MemoryPressure GetMemoryPressure()
-        {
-            const double HighPressureThreshold = .90;       // Percent of GC memory pressure threshold we consider "high"
-            const double MediumPressureThreshold = .70;     // Percent of GC memory pressure threshold we consider "medium"
-
-            GCMemoryInfo memoryInfo = GC.GetGCMemoryInfo();
-            if (memoryInfo.MemoryLoadBytes >= memoryInfo.HighMemoryLoadThresholdBytes * HighPressureThreshold)
-            {
-                return MemoryPressure.High;
-            }
-            else if (memoryInfo.MemoryLoadBytes >= memoryInfo.HighMemoryLoadThresholdBytes * MediumPressureThreshold)
-            {
-                return MemoryPressure.Medium;
-            }
-            return MemoryPressure.Low;
-        }
-
-        private static bool GetTrimBuffers()
-        {
-            // Environment uses ArrayPool, so we have to hit the API directly.
-#if !CORECLR
-            // P/Invokes are different for CoreCLR/RT- for RT we'll not allow
-            // enabling/disabling for now.
-            return true;
-#else
-            return CLRConfig.GetBoolValueWithFallbacks("System.Buffers.ArrayPool.TrimShared", "DOTNET_SYSTEM_BUFFERS_ARRAYPOOL_TRIMSHARED", defaultValue: true);
-#endif
+            return ((TlsOverPerCoreLockedStacksArrayPool<T>)target).Trim();
         }
 
         /// <summary>
         /// Stores a set of stacks of arrays, with one stack per core.
         /// </summary>
         private sealed class PerCoreLockedStacks
         {
+            /// <summary>Number of locked stacks to employ.</summary>
+            private static readonly int s_lockedStackCount = Math.Min(Environment.ProcessorCount, MaxPerCorePerArraySizeStacks);
+#if TARGET_64BIT
+            /// <summary>Multiplier used to enable faster % operations on 64-bit.</summary>
+            private static readonly ulong s_fastModMultiplier = HashHelpers.GetFastModMultiplier((uint)s_lockedStackCount);
```

Comment:
How much of the speedup are you getting thanks to this? To me, this feels we are going overboard here.

This is fallback path already and the hardware divider should be pretty fast for the small numbers that we are dealing with here.

Reply:
It's small.  I can revert it.  Seemed like "why not", but not a big deal.

Reply:
I would rather keep it simple.